contracts/CompoundGovernor.sol

```diff
@@ -64,6 +114,69 @@ contract CompoundGovernor is
         __GovernorPreventLateQuorum_init(_initialVoteExtension);
         __GovernorSettableFixedQuorum_init(_quorumVotes);
         __Ownable_init(_initialOwner);
+        _setWhitelistGuardian(_whitelistGuardian);
+        _setProposalGuardian(_proposalGuardian);
+    }
+
+    /// @notice Sets or updates the whitelist expiration for a specific account.
+    /// @param _account The address of the account to be whitelisted.
+    /// @param _expiration The timestamp until which the account will be whitelisted.
+    /// @dev Only the executor (timelock) or the whitelistGuardian can call this function.
+    function setWhitelistAccountExpiration(address _account, uint256 _expiration) external {
+        if (msg.sender != _executor() && msg.sender != whitelistGuardian) {
+            revert Unauthorized("Not timelock or guardian", msg.sender);
+        }
+
+        whitelistAccountExpirations[_account] = _expiration;
+        emit WhitelistAccountExpirationSet(_account, _expiration);
+    }
+
+    /// @notice Checks if an account is currently whitelisted.
+    /// @param _account The address of the account to check.
+    /// @return bool Returns true if the account is whitelisted (expiration is in the future), false otherwise.
+    function isWhitelisted(address _account) public view returns (bool) {
+        return (whitelistAccountExpirations[_account] > block.timestamp);
+    }
+
+    /// @notice Sets a new whitelistGuardian.
+    /// @param _newWhitelistGuardian The address of the new whitelistGuardian.
+    /// @dev Only the executor (timelock) can call this function.
+    function setWhitelistGuardian(address _newWhitelistGuardian) public {
+        if (msg.sender != _executor()) {
```

Comment:
Should I use the stricter `_checkGovernance` instead which checks if a proposal has actually passed beyond checking the `msg.sender`?

Reply:
https://github.com/ScopeLift/compound-governance-upgrade/pull/29/commits/3f5ee9512ff4a4699235bc41504c5559a83e7bee